blocks/monitoring/aggregation.py

```diff
@@ -151,6 +159,15 @@ def mean(numerator, denominator=1.):
     return variable
 
 
+def perplexity(log_likelihood, n_examples):
+    """Perplexity for total log_likelihood of n_examples."""
+    variable = tensor.exp(-log_likelihood / n_examples)
+    variable.tag.aggregation_scheme = Perplexity(log_likelihood, n_examples)
+    # There is no good default name for the output variable,
+    # so we don't try set any.
```

Comment:
At first I thought and it would be not nice to return always the same name. But maybe it's an okay default. Will change.